app/Pressure.php

```diff
@@ -12,17 +12,29 @@ public static function loadData($from = null, $to = null)
     {
         $result = null;
         if ($from != '' && $to != '') {
-            if (self::getSettingMaxValuesPerGraph() <= self::getCountRowsByDate($from, $to)) {
-                $result = self::getOptimizedDataByDate($from, $to);
-            } else {
-                $result = self::getDataByDate($from, $to);
-            }
+            $result = self::loadDataWithDate($from, $to);
         } else {
-            if (self::getSettingMaxValuesPerGraph() <= self::getCountRows()) {
-                $result = self::whereRaw('id mod ' . self::getNthRows(self::getCountRows()) . ' = 0')->get();
-            } else {
-                $result = self::get();
-            }
+            $result = self::loadDataWithoutDate();
+        }
+        return $result;
+    }
+
+    private static function loadDataWithDate($from, $to)
```

Comment:
loadDataWithDate and loadDatawithoutDate functions can be merge into with optional parameter which accept DateTime object, also these two function can abstracted to parent class, they are duplicated three times